### PR TITLE
IC-1170: Add Stylelint for SASS linting

### PR DESCRIPTION
**Pull request by Gweaton**
## What does this pull request do?

- Updates package-lock version for latest version of NPM (7.5.4 at time of PR)
- Adds Stylelint with a few custom rules to override the defaults (see below)
- Fixes existing linting violations (or adds rules to ignore them) 
- Adds automatic fixing of style errors to pre-commit hook

### Rules

- `"selector-class-pattern": null`

Allows us to use styles with double underscores.

- `"color-named": null`

Allows us to use named colours.

- `"Rule-empty-line-before": "always-multi-line"`

Ensures we have an empty line between each rule, except the first nested rule.

- `Max-nesting-depth: 2`

Set due to current css styles, I don't know what a sensible max nesting depth
is, but this at least allows our current implementation to pass.

- `selector-max-id: 1`

Allows us to apply styles to an ID - this is turned off in Stylelint as a
default, but our current styles from the Typescript template already apply
styles to IDs.

### Possibly controversial rule added:

Alphabetical ordering of styles - I do quite like this, as it keeps things neat and consistent, but can imagine others might prefer not to bother with this. However, as it's fixed automatically from now on, it shouldn't be a problem.

## What is the intent behind these changes?

- To enforce consistent SASS style throughout codebase.
- To save time on code review by ensuring code style violations are all caught automatically.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,8 +1,15952 @@
 {
```
Comment (lawrence-forooghian) [inline package-lock.json:1]: I'm finding this diff quite difficult to parse. What was the command that you ran to generate this change? Might be worth popping in the commit message.

File: package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "int-test-ui": "cypress open",
     "record-build-info": "node ./bin/record-build-info",
     "security_audit": "npx audit-ci --config audit-ci.json",
-    "lint": "eslint . --cache --max-warnings 0",
+    "lint": "eslint . --cache --max-warnings 0 && stylelint **/*.scss",
```
Comment (lawrence-forooghian) [inline package.json:25]: Will Stylelint know to exclude the files in `node_modules`?

File: doc/architecture/decisions/0010-use-stylelint-for-linting-styles.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# 10. Use Stylelint for linting styles
```
Comment (lawrence-forooghian) [inline doc/architecture/decisions/0010-use-stylelint-for-linting-styles.md:1]: Awesome

File: .circleci/config.yml [hunk 4]
```diff
@@ -26,7 +26,6 @@ jobs:
       tag: 14-stretch-browsers
     steps:
       - checkout
-      - node/install-npm
```
Comment (lawrence-forooghian) [inline .circleci/config.yml:29]: What motivated you to make this change?

**Review by lawrence-forooghian — COMMENTED** [review]
Awesome, great idea to add this. A few comments.

File: package.json (hunk 2 above)
Comment (Gweaton) [reply package.json:25]: Apparently, yes - according to [this](https://stylelint.io/user-guide/configure#ignorefiles):

> stylelint ignores the node_modules directory by default. However, this is overridden if ignoreFiles is set.

File: package-lock.json (hunk 1 above)
Comment (Gweaton) [reply package-lock.json:1]: Have added to the commit message, but it was running `npm install` with the latest stable version of npm - I'm not sure why it changed so much, but it seems to do so every time I run `npm install` (and did on one of Israt's branches too) so thought it'd be good to update here at least. I kept it in its own commit (separate from the one that adds the Stylelint package) to make it a bit clearer that it's separate.

File: .circleci/config.yml (hunk 4 above)
Comment (Gweaton) [reply .circleci/config.yml:29]: The build was failing on CI due to issues installing Cypress. I thought this was due to having multiple versions of npm or Cypress installed but I'm not sure if this is actually the correct fix - was hoping someone might be able to give me a hand with this today.

Comment (Gweaton) [reply .circleci/config.yml]: Have removed this to spend some more time on it today.

**Review by tomsmyers — APPROVED** [review]

**Pull request merged**